shared_model/backend/protobuf/common_objects/noncopyable_proto.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_NONCOPYABLE_PROTO_HPP
+#define IROHA_NONCOPYABLE_PROTO_HPP
+
+/**
+ * Generic class for handling proto objects which are not intended to be copied.
+ * @tparam Iface is interface to inherit from
+ * @tparam Proto is protobuf container
+ * @tparam Impl is implementation of Iface
+ */
+template <typename Iface, typename Proto, typename Impl>
+class NonCopyableProto : public Iface {
+ public:
+  using TransportType = Proto;
+
+  template <typename Transport>
```

Comment:
Add use case of ctor. The documentation should contain information about copy/move semantic of ctor.